### ISSUE TITLE
Support empty prefix

### ISSUE THREAD

**Issue by rnemykin** [spring-cloud/spring-cloud-consul#308]
Hi, i store in consul key smth like this
```
image.color.red=RED
image.color.green=GREEN
```
So many applications put such key/value to consul (A,B,C)and i want watch for them (for kyes with prefix _image.color.*_) from another application (D).
Currently i have to store properies like
```image/color/green``` because _org.springframework.cloud.consul.config.ConsulPropertySourceLocator_ works with '/' separator.

```
List<String> suffixes = new ArrayList<>();
if (this.properties.getFormat() != FILES) {
	suffixes.add("/");
} else {
	suffixes.add(".yml");
	suffixes.add(".yaml");
	suffixes.add(".properties");
}

String defaultContext = prefix + "/" + this.properties.getDefaultContext();
for (String suffix : suffixes) {
	this.contexts.add(defaultContext + suffix);
}
for (String suffix : suffixes) {
	addProfiles(this.contexts, defaultContext, profiles, suffix);
}
```

Also i would like to configure only key prefix 
`spring.cloud.consul.config.prefix=image.color`

and don't set properties like defaultContext, but it's _@NotEmpty_ and i can't set it to null (default is "application" which also brings inconvenience).
Similarly situation with _name_ property of _org.springframework.cloud.consul.config.ConsulConfigProperties_

May be intersect with
https://github.com/spring-cloud/spring-cloud-consul/issues/303

**Comment by spencergibb** [comment]
Not sure this make sense. The property sources are already watched and a spring event `EnvironmentChangeEvent` is published with the keys that have changed.

**Comment by rnemykin** [comment]
it makes some discomfort, if I specify 
`spring:
  cloud:
    consul:
      config:
        prefix: image.color`
, I expect that changing these keys (keys with such prefix) will update be handled, but it isn't.
Actually, with such config, spring do watching for keys
`[name='image.color/{spring.application.name}/'], ConsulPropertySource [name='image.color/application/']`

Because contexts for key watching building with some defaults values (defaultContext, name) and also separates with `/`. 

Until I looked in the source code I do not understand how can I set the configuration to get it working,
In docs (http://cloud.spring.io/spring-cloud-static/Dalston.RELEASE/#spring-cloud-consul-config-watch) nothing is said about it.

I would like the flexibility to configure such things

**Comment by rnemykin** [comment]
up

**Comment by spencergibb** [comment]
The prefix is a 'directory' or path in the consul heirarchy.

**Comment by pjhampton** [comment]
Yeah, +1 for this. My team have ran into a brick wall because of this and non-spring boot applications not using team-wide naming conventions. I don't see the need for the `@notempty` annotation - it's restrictive and boxes users in.

**Comment by spencergibb** [comment]
@pjhampton no, there is not. PRs welcome.

**Comment by pjhampton** [comment]
Sure, I can submit a PR :-)

**Comment by pjhampton** [comment]
@spencergibb I have a PR ready to go. Should I open 2 PRs? One against 1.3.x and 2.x?

**Comment by spencergibb** [comment]
Just one against 1.3.x and we'll merge forward

**Comment by spencergibb** [comment]
Closed via #392
